integration_tests/src/main/python/orc_test.py

```diff
@@ -663,3 +663,15 @@ def test_read_type_casting_integral(spark_tmp_path, offset, reader_confs, v1_ena
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark: spark.read.schema(rs).orc(data_path),
         conf=all_confs)
+
+def test_orc_read_count(spark_tmp_path):
+    data_path = spark_tmp_path + '/ORC_DATA'
+    gen_list = [('c' + str(i), gen) for i, gen in enumerate(orc_basic_gens)]
```

Comment:
Only tested basic ORC types, need to test more nested types like `array` `map` and `struct`.

Reply:
It's a nit, seems the `count` action does not scan the data.